var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -141,6 +146,22 @@ class Cmake(Package):
 
     phases = ['bootstrap', 'build', 'install']
 
+    @classmethod
+    def determine_spec_details(cls, prefix, exes_in_prefix):
+        exe_to_path = dict(
+            (os.path.basename(p), p) for p in exes_in_prefix
+        )
+        if 'cmake' not in exe_to_path:
+            return None
+
+        cmake = spack.util.executable.Executable(exe_to_path['cmake'])
+        output = cmake('--version', output=str)
+        if output:
+            match = re.search(r'cmake.*version\s+(\S+)', output)
+            if match:
+                version_str = match.group(1)
+                return Spec('cmake@{0}'.format(version_str))
```

Comment:
Maybe we can just:
```suggestion
                return 'cmake@{0}'.format(version_str)
```
and move the logic that parses the string into Specs in the command, as you suggested in a comment above.